pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -722,6 +727,22 @@ func ReadShootedSeed(shoot *gardencorev1beta1.Shoot) (*ShootedSeed, error) {
 	return shootedSeed, nil
 }
 
+// GetManagedSeed gets the ManagedSeed resource for this Shoot. If Spec.ManagedSeedName is nil or a ManagedSeed resource
+// with the specified name is not found, nil is returned.
+func GetManagedSeed(ctx context.Context, c client.Client, shoot *gardencorev1beta1.Shoot) (*seedmanagementv1alpha1.ManagedSeed, error) {
```

Comment:
See above, this is intentionally part of the spec and not e.g. an annotation, because we want to trigger a reconciliation every time it changes, in order to apply the new API server settings. It has nothing to do with lookup.